### ISSUE TITLE
bug(datadog_metrics sink): rate counters being sent as counts (again)

### ISSUE THREAD

**Issue by tobz** [vectordotdev/vector#14740]
In #13394, we added the ability for the `datadog_metrics` sink to properly send "rate" metrics to Datadog, which describe a counter in terms of its total value over a given interval.

In #13960, while fixing another closely-related issue, I mistaken broke the logic introduced in #13394 such that regardless of whether or not an interval was present, we were sending all counters as "count" metrics.

Long story short, this sends those rate metrics in a way that makes them incorrect when compared to the same metric being sent directly by the Datadog Agent. We need to reapply the logic that sends counters as a rate if they have an interval.
